gpuflow/dataplane/gpu/cuda/cuda_lpm_factory.cu

```diff
@@ -160,7 +160,13 @@ int IPv6LPMFactory::AddLPMRule(uint8_t *ipv6_address, uint8_t depth, uint8_t nex
 
   }
 
-  SetupIPv6RuleEntry<<<1, end -start>>>(IPv6TBL24, start, next_hop, depth);
+  if (end < start){
```

Comment:
How's this occurred?

Reply:
When ipv6_addr[2], ipv6_addr[1] and ipv6_addr[2] all equal 255.

Reply:
I see, but it's better bound checking before end is going to be overflow.

Reply:
REMARK: 

Actaully, we don't need to have `end` variable. The only needed is distance.
The `start` and `end` for my previous implementation is used for bit operation for calculating distance.
Instead, the way you are using for loop for calculating distance is fine, but the `end` is meaningless now.
Therefore, you can bound checking like this,
```c++
if (start + distance > (1<< 24)) {
  // overflow
  distance = (1 << 24) - start;
}
```

Reply:
Further more, you can reduce the repetition of distance calculating to upper, before checking depth's condition.